src/components/LocaleLink.js

```diff
@@ -4,42 +4,53 @@ import { Link, StaticQuery, graphql } from "gatsby"
 
 import { withLocale } from "./App"
 
-export const CustomLocaleLink = ({ children, to, locale, ...other }) =>
-  <StaticQuery
-    query={query}
-    render={({
-      site: {
-        siteMetadata: {
-          defaultLocale,
-          supportedLocales,
+export const CustomLocaleLink = ({ children, to, locale, ...other }) => {
+  if (/^http/.test(to)) {
```

Comment:
You can use `to.startsWith("http")` here for more clarity/marginal speed. Not a huge deal, just sharing it.

Reply:
Nice, that's much better!